crates/polars-plan/src/plans/aexpr/schema.rs

```diff
@@ -84,11 +84,14 @@ impl AExpr {
                     .get(*expr)
                     .to_field_impl(schema, ctx, arena, &mut false)?;
 
-                if let List(inner) = field.dtype() {
-                    Ok(Field::new(field.name().clone(), *inner.clone()))
-                } else {
-                    Ok(field)
-                }
+                let field = match field.dtype() {
+                    List(inner) | Array(inner, ..) => {
```

Comment:
Fix for the linked issue, we didn't have the correct lazy schema on an exploded `Array` column